### PR TITLE
shell completion for kustomize commands

### LINKED ISSUE #1788: Autocomplete kyaml subcommand names
The subcommands for `kyaml` should support bash auto completion.

### PR DESCRIPTION
**Pull request by pwittrock**
fixes #1788

### CODE REVIEW

File: cmd/config/cmd/shell-complete/shell-complete.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package shell_complete
```
Comment (monopole) [inline cmd/config/cmd/shell-complete/shell-complete.go]: can we avoid the snake case and here to be consistent w/ package names?
the underscore in package name has special meaning sometimes (_test)

File: cmd/config/cmddocs/commands/docs.go [hunk 2]
```diff
@@ -21,6 +21,16 @@ var CatExamples = `
     # unwrap Resource config from a directory in an ResourceList
     ... | kustomize config cat`
 
+var CompletionShort = `Install shell completion for kustomize commands and flags`
+var CompletionLong = `
+Install shell completion for kustomize commands and flags.
```
Comment (monopole) [inline cmd/config/cmddocs/commands/docs.go]: can this say what the consequences of the _install_ are?
local artifacts?
when would i want/need to reinstall?

File: cmd/config/cmd/shell-complete/shell-complete.go (hunk 1 above)
Comment (pwittrock) [reply cmd/config/cmd/shell-complete/shell-complete.go]: Done

File: cmd/config/cmddocs/commands/docs.go (hunk 2 above)
Comment (pwittrock) [reply cmd/config/cmddocs/commands/docs.go]: Done

**Comment by pwittrock** [comment]
PTAL

**Pull request merged**